plugins/org.jkiss.dbeaver.ui.editors.sql/src/org/jkiss/dbeaver/ui/editors/sql/handlers/SQLEditorHandlerOpenEditor.java

```diff
@@ -274,21 +274,64 @@ public static SQLEditor openSQLConsole(
         return openSQLConsole(workbenchWindow, context, sqlInput);
     }
 
-    public static SQLEditor openSQLConsole(IWorkbenchWindow workbenchWindow, SQLNavigatorContext context, IEditorInput sqlInput) {
+    public static SQLEditor openSQLConsole(
+        @NotNull IWorkbenchWindow workbenchWindow,
+        @NotNull SQLNavigatorContext context,
+        @NotNull IEditorInput sqlInput
+    ) {
         EditorUtils.setInputDataSource(sqlInput, context);
         return openSQLEditor(workbenchWindow, sqlInput);
     }
 
+    /**
+     * Open a new SQL console editor.
+     * <p>
+     * This method always opens a new editor instance even if an editor with the same input
+     * is already opened. The provided {@code sqlInput} will be associated with the
+     * supplied SQLNavigatorContext using EditorUtils, and the editor will be opened with
+     * {@link IWorkbenchPage#MATCH_NONE} to avoid reusing existing editors.
```

Comment:
Seems redundant as it explains its inner workings/implementation details


```suggestion
     * is already opened. The provided {@code sqlInput} will be associated with the
     * supplied {@code context}.
```